### PR TITLE
feat(TDS-6666): add skipping commas with back slash

### PR DESCRIPTION
**Pull request by abdelillah-tech**
**What is the problem this PR is trying to solve?**

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js [hunk 1]
```diff
@@ -72,6 +77,7 @@ class EnumerationForm extends React.Component {
 		this.allowImport = false;
 		const disabledAction = props.schema ? props.schema.disabled : false;
 		this.importFileHandler = this.importFileHandler.bind(this);
+		this.enableSkip = props.enableSkip;
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js:80]: if the value change later on this property will not be updated

**Review by jmfrancois — CHANGES_REQUESTED** [review]

File: packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js (hunk 1 above)
Comment (jmfrancois) [reply packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js:80]:
```suggestion
```

File: packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js [hunk 2]
```diff
@@ -432,7 +438,7 @@ class EnumerationForm extends React.Component {
 					actionsEdit: this.loadingInputsActions,
 				},
 			}));
-			const formattedValue = EnumerationForm.parseStringValueToArray(value.value);
+			const formattedValue = EnumerationForm.parseStringValueToArray(value.value, this.enableSkip);
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js:441]:
```suggestion
			const formattedValue = EnumerationForm.parseStringValueToArray(value.value, this.props.enableSkip);
```

File: packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js [hunk 3]
```diff
@@ -467,9 +473,10 @@ class EnumerationForm extends React.Component {
 			}
 			item.displayMode = enumerationStates.DISPLAY_MODE_DEFAULT;
 			const valueExist = this.valueAlreadyExist(value.value, this.state);
+
 			// if the value is empty, no value update is done
 			if (value.value && !valueExist) {
-				item.values = EnumerationForm.parseStringValueToArray(value.value);
+				item.values = EnumerationForm.parseStringValueToArray(value.value, this.enableSkip);
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js:479]:
```suggestion
				item.values = EnumerationForm.parseStringValueToArray(value.value, this.props.enableSkip);
```

File: packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js [hunk 4]
```diff
@@ -698,10 +705,11 @@ class EnumerationForm extends React.Component {
 			this.setState({
 				headerInput: this.loadingInputsActions,
 			});
+
 			this.props
 				.onTrigger(event, {
 					trigger: {
-						value: EnumerationForm.parseStringValueToArray(value.value),
+						value: EnumerationForm.parseStringValueToArray(value.value, this.enableSkip),
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js:712]:
```suggestion
						value: EnumerationForm.parseStringValueToArray(value.value, this.props.enableSkip),
```

File: packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js [hunk 5]
```diff
@@ -725,7 +733,7 @@ class EnumerationForm extends React.Component {
 				schema,
 				value: this.state.items.concat([
 					{
-						values: EnumerationForm.parseStringValueToArray(value.value),
+						values: EnumerationForm.parseStringValueToArray(value.value, this.enableSkip),
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js:736]:
```suggestion
						values: EnumerationForm.parseStringValueToArray(value.value, this.props.enableSkip),
```

File: packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js [hunk 6]
```diff
@@ -49,7 +49,12 @@ class EnumerationForm extends React.Component {
 		return ITEMS_DEFAULT_HEIGHT;
 	}
 
-	static parseStringValueToArray(values) {
+	static parseStringValueToArray(values, enableSkip) {
```
Comment (Kporal) [inline packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js:52]: could you add some test for this please?

File: packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js [hunk 7]
```diff
@@ -1050,6 +1057,7 @@ if (process.env.NODE_ENV !== 'production') {
 		onTrigger: PropTypes.func.isRequired,
 		properties: PropTypes.object,
 		schema: PropTypes.object,
+		enableSkip: PropTypes.bool,
```
Comment (Kporal) [inline packages/forms/src/UIForm/fields/Enumeration/EnumerationWidget.js:1060]: The name is not clear enough, I suggest a rename, don't know if you can find better?
```suggestion
		skipCommas: PropTypes.bool,
```

**Review by Kporal — APPROVED** [review]

**Review by jmfrancois — APPROVED** [review]

**Pull request merged**